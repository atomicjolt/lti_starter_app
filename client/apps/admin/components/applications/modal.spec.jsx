import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import ReactTestUtils, { act } from 'react-dom/test-utils';
import Modal from './modal';

describe('applications modal', () => {
  let result;
  let saved;

  const application = {
    name: 'SPEC_NAME',
    description: 'SPEC_STRING'
  };
  const isOpen = true;
  const closeModal = () => {};
  const save = () => { saved = true; };

  // https://medium.com/@amanverma.dev/mocking-create-portal-to-utilize-react-test-renderer-in-writing-snapshot-uts-c49773c88acd
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element, node) => element);
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });

  beforeEach(() => {
    saved = false;

    result = TestRenderer.create(
      <Modal
        application={application}
        isOpen={isOpen}
        closeModal={closeModal}
        save={save}
      />);
  });

  it('modal matches snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('saves the application', () => {
    expect(saved).toBeFalsy();

    const container = document.createElement('div');
    document.body.appendChild(container);
    act(() => {
      ReactDOM.render(
        <Modal
          application={application}
          isOpen={isOpen}
          closeModal={closeModal}
          save={save}
        />,
        container
      );
    });
    const buttons = document.getElementsByTagName('button');
    expect(buttons.length).toEqual(2);
    const button = buttons[0];
    ReactTestUtils.Simulate.click(button);

    expect(saved).toBeTruthy();
  });
});
