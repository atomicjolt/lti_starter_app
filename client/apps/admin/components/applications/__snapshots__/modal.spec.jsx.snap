// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications modal modal matches snapshot 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="c-modal c-modal--settings is-open"
  closeTimeoutMS={0}
  contentElement={[Function]}
  contentLabel="Modal"
  isOpen={true}
  onRequestClose={[Function]}
  overlayClassName="c-modal__background"
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  save={[Function]}
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h2
    className="c-modal__title"
  >
    SPEC_NAME
     Settings
  </h2>
  <Form
    closeModal={[Function]}
    configParseError={null}
    description="SPEC_STRING"
    onChange={[Function]}
    save={[Function]}
  />
</Modal>
`;
